src/utils/jwt.ts

```diff
@@ -0,0 +1,126 @@
+import canonicalize from 'canonicalize';
+import AccountBase from '../account/Base';
+import {
+  Encoded, Encoding, decode, encode,
+} from './encoder';
+import { verify } from './crypto';
+import { ArgumentError, InvalidSignatureError } from './errors';
+
+// TODO: use Buffer.from(data, 'base64url') after solving https://github.com/feross/buffer/issues/309
+const toBase64Url = (data: Buffer | Uint8Array | string): string => Buffer
+  .from(data)
+  .toString('base64')
+  .replaceAll('/', '_')
+  .replaceAll('+', '-')
+  .replace(/=+$/, '');
+
+const fromBase64Url = (data: string): Buffer => Buffer
+  .from(data.replaceAll('_', '/').replaceAll('-', '+'), 'base64');
+
+const objectToBase64Url = (data: any): string => toBase64Url(canonicalize(data) ?? '');
+
+const header = 'eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9'; // objectToBase64Url({ alg: 'EdDSA', typ: 'JWT' })
+
+/**
+ * JWT including specific header
+ * @category JWT
+ */
+export type Jwt = `${typeof header}.${string}.${string}`;
+
+/**
+ * Generate a signed JWT
+ * Provide `"sub_jwk": undefined` in payload to omit signer public key added by default.
+ * @param originalPayload - Payload to sign
+ * @param account - Account to sign by
+ * @category JWT
+ */
+export async function signJwt(originalPayload: any, account: AccountBase): Promise<Jwt> {
+  const payload = { ...originalPayload };
+  if (!('sub_jwk' in payload)) {
+    payload.sub_jwk = {
+      kty: 'OKP',
+      crv: 'Ed25519',
+      x: toBase64Url(decode(account.address)),
```

Comment:
https://datatracker.ietf.org/doc/html/draft-ietf-jose-cfrg-curves-06#section-2
is it based on something like this? In the next chapter "Ed25519" defined, but I can't find it in "JSON Web Elliptic Curve" registry.
Initially I found "sub_jwk" in https://www.iana.org/assignments/jwt/jwt.xhtml